waimak-core/src/main/scala/com/coxautodata/waimak/dataflow/DataFlowExecutor.scala

```diff
@@ -14,28 +17,115 @@ trait DataFlowExecutor[C] {
     * @return (Seq[EXECUTED ACTIONS], FINAL STATE). Final state does not contain the executed actions and the outputs
     *         of the executed actions are now in the inputs
     */
-  def execute(dataFlow: DataFlow[C]): (Seq[DataFlowAction[C]], DataFlow[C])
+  def execute(dataFlow: DataFlow[C]): (Seq[DataFlowAction[C]], DataFlow[C]) = {
+    val preparedDataFlow = dataFlow.prepareForExecution()
+
+    val executionResults: Try[(Seq[DataFlowAction[C]], DataFlow[C])] = loopExecution(preparedDataFlow, initActionScheduler(), Seq.empty)
+
+    executionResults.get
```

Comment:
Calling `.get` on a Failure will cause the application to throw an exception saying that you can't call .get on a Failure, meaning we lose the actual reason for the Failure. This will need to be unpacked properly